### PR TITLE
[SERVICE-800] Create "async utils" module

### PR DESCRIPTION
**Pull request by bryangaleopenfin**

### CODE REVIEW

**Review by ottah — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "url": "https://github.com/HadoukenIO/service-async"
+  },
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/HadoukenIO/service-async/issues"
+  },
+  "homepage": "https://github.com/HadoukenIO/service-async",
+  "devDependencies": {
+    "@types/openfin": "^43.0.1",
```
Comment (tomrobertsOF) [inline package.json]: Don't think you need this dependency right?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "fs-extra": "^8.1.0",
+    "openfin-service-signal": "^1.0.0",
+    "openfin-service-tooling": "^1.0.22",
```
Comment (tomrobertsOF) [inline package.json:30]: You have this as a dev and peer dependency. Is that correct?

File: package.json (hunk 1 above)
Comment (tomrobertsOF) [reply package.json]: Also the js-adapter isn't used anywhere that I can see.

File: test/jest-unit.config.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (tomrobertsOF) [inline test/jest-unit.config.json]: Any reason to have this empty file here?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "check": "svc-tools check",
+    "clean": "rimraf dist",
+    "build": "npm run clean && tsc -p tsconfig.json && node src/fileCopy.js",
+    "deploy:local": "npm run build && cd dist && npm pack",
+    "fix": "svc-tools fix",
+    "test": "npm run test:unit",
+    "test:unit": "svc-tools test unit"
```
Comment (tomrobertsOF) [inline package.json:13]: Do we really need to use service tools for this? Does it give us anything more than just a bog standard jest run?

**Review by tomrobertsOF — COMMENTED** [review]
A couple of minor comments. Not too committed to any, so feel free to push back on anything.

File: package.json (hunk 4 above)
Comment (bryangaleopenfin) [reply package.json:13]: You forgot linting! 😛

File: test/jest-unit.config.json (hunk 3 above)
Comment (bryangaleopenfin) [reply test/jest-unit.config.json]: Ah, was assuming that something somewhere was assuming the presence of this file, turns out not. Deleted

File: package.json (hunk 1 above)
Comment (bryangaleopenfin) [reply package.json]: `build` fails without `@types/openfin`, `fix`/`check` fails without `hadouken-js-adapter`

File: package.json (hunk 2 above)
Comment (bryangaleopenfin) [reply package.json:30]: Seems no!

File: package.json (hunk 4 above)
Comment (tomrobertsOF) [reply package.json:13]: Ah, of course

File: package.json (hunk 1 above)
Comment (tomrobertsOF) [reply package.json]: Hmmm. That implies that they should be a sub dependency of something, since they're not used directly.

Comment (bryangaleopenfin) [reply package.json]: It's service-tooling that needs them, which has them as dev dependencies. FDC3 and Notifications also have them as dev dependencies and uses them directly.

Is suspect how things are now may be down to weirdness around a package both directly and indirectly depending on two different versions of the same package, but will come back to in the morning

File: package.json (hunk 4 above)
Comment (ottah) [reply package.json:13]: Does it make sense to pull out linting into a package so we can use it in our packages?

**Review by tomrobertsOF — APPROVED** [review]

**Pull request merged**